examples/quickstart/jupyter-notebooks/druidapi/sql.py

```diff
@@ -0,0 +1,690 @@
+            request = self.sql_request(request)
+        if is_blank(request.sql):
+            raise ClientError("No query provided.")
+        if self.rest_client().trace:
+            print(request.sql)
+        query_obj = request.to_request()
+        return (request, query_obj)
+
+    def sql_query(self, request) -> SqlQueryResult:
+        '''
+        Submit a SQL query with control over the context, parameters and other
+        options. Returns a response with either a detailed error message, or
+        the rows and query ID.
+        '''
+        request, query_obj = self._prepare_query(request)
+        r = self.rest_client().post_only_json(REQ_ROUTER_SQL, query_obj, headers=request.headers)
+        return SqlQueryResult(request, r)
+
+    def sql(self, sql, *args):
+        if len(args) > 0:
+            sql = sql.result_format(*args)
+        resp = self.sql_query(sql)
+        if resp.ok():
+            return resp.rows()
+        raise ClientError(resp.error_msg())
+
+    def explain_sql(self, query):
+        """
+        Run an EXPLAIN PLAN FOR query for the given query.
+
+        Returns
+        -------
+        An object with the plan JSON parsed into Python objects:
+        plan: the query plan
+        columns: column schema
+        tables: dictionary of name/type pairs
+        """
+        if is_blank(query):
+            raise ClientError("No query provided.")
+        results = self.sql('EXPLAIN PLAN FOR ' + query)
+        return results[0]
+
+    def sql_request(self, sql):
+        return SqlRequest(self, sql)
+
+    def show(self, query):
+        result = self.sql_query(query)
+        if result.ok():
+            result.show()
+        else:
+            display.display.show_error(result.error_msg())
+
+    def task(self, request):
+        request, query_obj = self._prepare_query(request)
+        r = self.rest_client().post_only_json(REQ_ROUTER_SQL_TASK, query_obj, headers=request.headers)
+        return QueryTaskResult(request, r)
+
+    def run_task(self, request):
+        resp = self.task(request)
+        if not resp.ok():
+            raise ClientError(resp.error_msg())
+        resp.wait_done()
+
+    def _tables_query(self, schema):
+        return self.sql_query('''
+            SELECT TABLE_NAME AS TableName
+            FROM INFORMATION_SCHEMA.TABLES
+            WHERE TABLE_SCHEMA = '{}'
+            ORDER BY TABLE_NAME
+            '''.format(schema))
+
+    def tables(self, schema=consts.DRUID_SCHEMA):
+        return self._tables_query(schema).rows()
+
+    def show_tables(self, schema=consts.DRUID_SCHEMA):
+        self._tables_query(schema).show()
+
+    def _schemas_query(self):
+        return self.sql_query('''
+            SELECT SCHEMA_NAME AS SchemaName
+            FROM INFORMATION_SCHEMA.SCHEMATA
+            ORDER BY SCHEMA_NAME
+            ''')
+
+    def show_schemas(self):
+        self._schemas_query().show()
+
+    def describe_table(self, part1, part2=None):
```

Comment:
What is part1 and part2? Is this method only intended to be used internally? Same question on `describe_function()`